src/bn.js

```diff
@@ -10,27 +10,92 @@ const BN = () => BigNumber
 /**
  * @param {string|number|BigNumber} value
  */
-const nBN = (value) => new BigNumber(value)
+const nBN = (value) => {
+  if (Number.isNaN(value) || value === 'NaN') throw new Error('ERR_NUM_NAN')
+  if (typeof value !== 'string' && typeof value !== 'number' &&
+    !(value instanceof BigNumber)) throw new Error('ERR_NUM_NAN')
+  if (value === 'Infinity' || value === '-Infinity' ||
+    value === Infinity || value === -Infinity) throw new Error('ERR_NUM_INFINITE')
 
-/**
- * @param {string|number|BigNumber} value
- */
-const validateBN = (value, opts = { allowNegative: true, allowZero: true, allowDecimals: true }) => {
-  /** @type {BigNumber} */
-  let n
   try {
-    n = nBN(value)
+    return new BigNumber(value)
   } catch (err) {
     throw new Error('ERR_NUM_NAN')
   }
+}
 
+/**
+ * @param {string|number|BigNumber} value
+ */
+const validateBN = (value, opts = { allowNegative: true, allowZero: true, allowDecimals: true }) => {
+  const n = nBN(value)
   if (opts.allowZero === false && n.isZero()) throw new Error('ERR_NUM_ZERO')
   if (opts.allowNegative === false && n.isNegative()) throw new Error('ERR_NUM_NOT_POS')
   if (opts.allowDecimals === false && /e|\./.test(n.toString())) throw new Error('ERR_NUM_WITH_PRECISION')
 }
 
+/**
+ * @param {...(string|number|BigNumber)} values
+ * @returns {BigNumber}
+ */
+const safeSumBN = (...values) => {
```

Comment:
not sure if we need the additional wrappers for additons, substractions and multiplications, they add possible error sources

Reply:
removed the function